shard/key.go

```diff
@@ -30,6 +30,11 @@ func KeyFromCID(cid cid.Cid) Key {
 	return Key{str: cid.String()}
 }
 
+// KeyFromCIDMultihash returns a key based on the multihash of the given cid.
+func KeyFromCIDMultihash(cid cid.Cid) Key {
```

Comment:
Is it too hard to do this in the application? Everything else is a constructor. If anything, I'd make this a `KeyFromMultihash` and pass in the multihash directly.

Reply:
Removed this completely.